samza-core/src/main/scala/org/apache/samza/storage/ContainerStorageManager.java

```diff
@@ -838,15 +831,24 @@ private boolean awaitSideInputTasks() throws InterruptedException {
    * @return the task store.
    */
   public Optional<StorageEngine> getStore(TaskName taskName, String storeName) {
+    if (taskStores == null) {
+      throw new SamzaException(String.format(
+          "Attempting to access store %s for task %s before ContainerStorageManager is started.",
+          storeName, taskName));
+    }
```

Comment:
can we use lifecycle signal instead of null check to infer initialization complete?